src/ripple/json/json_writer.h

```diff
@@ -175,6 +175,36 @@ std::string valueToQuotedString ( const char* value );
 /// \see Json::operator>>()
 std::ostream& operator<< ( std::ostream&, const Value& root );
 
+//------------------------------------------------------------------------------
+
+using write_t = std::function<void(void const*, std::size_t)>;
+
+/** Stream compact JSON to the specified function. */
+void
+stream (Json::Value const& jv, write_t write);
```

Comment:
Don't take `write_t` by value. `std::function` may allocate, and copying around the `std::functions` may be expensive. One way to fix this is the take by `const&`. A second way would be to make `write_t` a template parameter so the lambdas may be passed in without using `std::functional`. If a template parameter is used, `write_value` and `write_string` need to be templates too and moved into the header (in the detail namespace, of course).

I'm OK if you prefer to just take the param by `const&` instead of making it a template. All three functions should be changed tho.

Reply:
This was just relocated from `json_value.h` I prefer the template approach and will check if there wasn't a good reason for not doing so initially.